app/controllers/api/v2/courses_controller.rb

```diff
@@ -102,6 +102,14 @@ def update
           @course.save
           @course.course_subjects.each(&:save)
 
+          if subject_ids
+            NotificationService::CourseSubjectsUpdated.call(
+              course: @course,
+              previous_subject_names: @previous_subject_names,
+              previous_course_name: @previous_course_name,
```

Comment:
Not strictly related to this line of code but why do you need to use the previous course name?

Reply:
Might be worth checking how @avinhurry & I did course title changed notification https://github.com/DFE-Digital/teacher-training-api/pull/1481. Could this approach perhaps be a better on instead of having another notification template, azure changes etc.

Reply:
As discussed - in the controller the courses subjects are being re-assigned (determined by what ever subject ids are passed in params). See https://github.com/DFE-Digital/teacher-training-api/pull/1478/files#diff-7d6b888ba704f4684c5b576a1dd07676R190

When updating subjects on a course the name is automatically updated to reflect the change in subjects. This is why the previous course name is being captured in an instance var and passed through to the service and mailer.